src/main/java/io/hydrox/inventorysummary/InventorySummaryPlugin.java

```diff
@@ -117,13 +123,23 @@ private void groupItems()
 
 	private boolean isWhitelisted(Item item)
 	{
-		if (config.whitelistEnabled()) {
-			ItemComposition itemComposition = itemManager.getItemComposition(item.getId());
-			return whitelist.contains(itemComposition.getName());
-		}
-
 		// If the whitelist isn't enabled, all items are allowed
-		return true;
+		if (!config.whitelistEnabled()) return true;
+
+		ItemComposition itemComposition = itemManager.getItemComposition(item.getId());
+		return whitelist.contains(itemComposition.getName());
+	}
+
+	private boolean isBlacklisted(Item item)
```

Comment:
This really doesn't need to be a second function, as the original function can be simplified down to 
```java
return (!config.whitelistEnabled() || whitelist.contains(itemComp.getName())) && !(config.blacklistEnabled() && blacklist.contains(itemComp.getName()))
```
(Removing the config check initially and just going straight for the ItemComposition)

I forgot how rusty I was at boolean maths, but I've tried it and it checks out. 

This does put the Whitelist as more important than the Blacklist; idk if that's desirable, if you're rather see the stuff you explicity want to see rather than not see what you explicitly don't.

Reply:
As it stands now, the blacklist takes precedence over the whitelist, which kind of makes sense to me. I considered which takes precedence configurable, but I think that it's probably better as something that would be quick to add if requested. 

The function can definitely be one-lined, but I decided to separate it out into functions for readability's sake. I agree that it can be kind of excessive to have two separate functions. I can merge them into a single, `shouldItemBeShown` function, or something along those lines if you think that'd be an improvement. Honestly I kind of like the clarity having the separate functions bring to the `groupItems` function.

Reply:
Two functions just feels excessive, since the function should basically be checking if the item should be shown, like you mention. The one liner is a way to avoid a bunch of bulky ifs, and yeah it's a bit obtuse, but idk, the result is technically intuitive with how the lists are supposed to work.

It may be worth looking at `Ground Items` for list order, since there's a similar decision there for Highlight vs Hidden, and idr which way that goes.

Reply:
Yea you're definitely suggesting something more along the lines of what `Ground Items` is doing. 

```java
private boolean shouldItemBeShown(Item item)
{
    ItemComposition itemComp = itemManager.getItemComposition(item.getId());
    return (
            !config.whitelistEnabled()
                    || whitelist.contains(itemComp.getName())
    ) && !(
            config.blacklistEnabled()
                    && blacklist.contains(itemComp.getName())
    );
}
```

The function is most definitely shorter, but in my opinion this is way, way, less unreadable. But, it's your repo so this can be the route we go down.

This is a shorter version I personally find way more readable.

```java
private boolean shouldItemBeShown(Item item)
{
    ItemComposition itemComp = itemManager.getItemComposition(item.getId());
    if (config.whitelistEnabled() && !config.whitelist().contains(itemComp.getName())) 
    {
        return false;
    } 
    if (config.blacklistEnabled() && config.blacklist().contains(itemComp.getName()))
    {
        return false;
    }
    return true;
}
```

Or perhaps, IntelliJ's suggested change to the the 2nd version:

```java
private boolean shouldItemBeShown(Item item)
{
    ItemComposition itemComp = itemManager.getItemComposition(item.getId());
    if (config.whitelistEnabled() && !config.whitelist().contains(itemComp.getName()))
    {
        return false;
    }
    return !config.blacklistEnabled() || !config.blacklist().contains(itemComp.getName());
}
```

Let me know what you think

Reply:
Personally prefer the single return without a bunch of linebreaks, just one linebreak after the first `&&` should be fine

Reply:
You got it. Changed in latest PR.